### DISCUSSION TITLE
Edge Functions 502: Bad Gateway Error

### DISCUSSION THREAD

**Discussion by benjaminbialy** [supabase/supabase#8187] [Questions] [answered]
Hey guys,

I updated my edge functions today and started facing the following error

```
502: Bad Gateway (DEPLOYMENT_FAILED)

The deployment failed while serving the request.
```

I'm only getting this error on the functions that have been changed, even after reverting to the older version before the issue started occurring. 

Is there something that I am missing? I doubt there's a memory issue as the functions will now be the same size that they were before the error.

Can someone point me in the right direction? Thanks!

Note: I'm also getting this error when trying to make a new function

```
Error: Failed to create a new Function on the Supabase project: {"id":id,"slug":"test_transactions","name":"test_transactions","version":1,"status":"ACTIVE","created_at":1660124404890,"updated_at":1660124404890,"verify_jwt":true}
```

**Comment by sarkarology** [comment]
I am facing the same error. Locally functions are working well, but deployment shows error.

And in the function's log, it shows the error:

"unable to connect to isolate, terminated"

And when trying to access the function through supabasejs client (supabase.functions.invoke()), the network request shows CORS error, while the cors is provided:


```
import { serve } from 'https://deno.land/std@0.131.0/http/server.ts'

const corsHeaders = {
  'Access-Control-Allow-Origin': '*',
  'Access-Control-Allow-Headers': 'authorization, x-client-info, apikey, Content-Type',
}

console.log(`Function "browser-with-cors" up and running!`)

serve(async (req) => {
  // This is needed if you're planning to invoke your function from a browser.
  if (req.method === 'OPTIONS') {
    return new Response('ok', { headers: corsHeaders })
  }

  try {
    const { name } = await req.json()
    const data = {
      message: `Hello ${name}!`,
    }

    return new Response(JSON.stringify(data), {
      headers: { ...corsHeaders, 'Content-Type': 'application/json' },
      status: 200,
    })
  } catch (error) {
    return new Response(JSON.stringify({ error: error.message }), {
      headers: { ...corsHeaders, 'Content-Type': 'application/json' },
      status: 400,
    })
  }
})
```

**Reply by benjaminbialy** [reply]
That's so odd, I'm getting "unable to connect to isolate, terminated" in my logs too.

The weirdest part is that the functions that I haven't touched are working, with no errors.

I'm very lost, hopefully this will get some attention, did this only start happening today for you too?

**Reply by sarkarology** [reply]
Yes, my current project is stuck due to this issue.

Tried everything from my end. For HOURS. Can't resolve.

Hope this gets some attention.

**Reply by GaryAustin1** [reply]
You all should file a support request if you have not.  There are Discord users reporting the same issue.

**Comment by soedirgo** [comment, marked as answer]
~~Hey folks, this is an issue on our side. Can you upgrade the CLI to v0.38.4+, redeploy the function, and try triggering the function again?~~

Sorry folks, this one's on us. Can you redeploy your Functions and try invoking it again?

**Reply by myemuk** [reply]
Hi, I've upgraded the CLI, restarted server, created new function and deployed it. Still same issue

**Reply by troels** [reply]
I've upgraded, redeployed and still same issue.

**Reply by paddyfink** [reply]
Same here

**Reply by grantv** [reply]
Same here

**Reply by soedirgo** [reply]
Can you redeploy your Functions and try invoking it again?

**Reply by myemuk** [reply]
Yes, it works now. Thanks a lot

**Reply by benjaminbialy** [reply]
> Can you redeploy your Functions and try invoking it again?

Legend. Thanks heaps, works now!
